### PR TITLE
Action management/walk synonym

### LINKED ISSUE #476: Reintroduce "WALK" Synonym for "GO" Action
As described by issue #120, several synonyms were removed from the action struct to fix a complication in creating structs in get_supported_actions function. This implements the "WALK" synonym for the "GO" action.  This involved determining which structs and functions are affected by this change, and implementing the appropriate compatibility.

### PR DESCRIPTION
**Pull request by isaacfink21**
Resolves #476, reintroducing the "WALK" synonym for "GO" action.

### CODE REVIEW

**Comment by isaacfink21** [comment]
As of sprint 2, this PR will resolve #120, reintroducing support for action synonyms that were previously removed due to complications with the `get_supported_actions()` function. The following synonyms are now supported:

```
PICKUP, // synonym for "take"
USE, 
DRINK,
EAT,    // synonyms for "consume"
WALK    // synonym for "go"
```

It also updates the documentation with the new actions.

File: src/action_management/src/get_actions.c [hunk 1]
```diff
@@ -15,12 +15,17 @@ static action_type_t valid_actions[] =
     {"TURNON", ITEM},
```
Comment (Dumplingz) [inline src/action_management/src/get_actions.c:15]: This looks good! However, I'm not sure what the BIN files are supposed to do (I checked master and dev and they don't have them, seems like the editor's files but I may be wrong).

File: src/action_management/src/get_actions.c [hunk 2]
```diff
@@ -15,12 +15,17 @@ static action_type_t valid_actions[] =
...
     {"TURNOFF", ITEM},
     {"TAKE", ITEM},
+    {"PICKUP", ITEM},
```
Comment (Dumplingz) [inline src/action_management/src/get_actions.c:18]: I see that you have declared "PICKUP" and other synonyms here as their own actions, but I'm having a hard time finding where they become functionally the same as the original "TAKE" action.

File: tests/action_management/test_get_actions.c [hunk 3]
```diff
@@ -65,9 +80,24 @@ Test(get_actions, search_success)
     cr_assert_eq(go->kind, PATH,
                  "Expected the action kind %d, but got action kind %d.\n",
                  PATH, go->kind);
-    cr_assert_eq(use->kind, ITEM_ITEM,
```
Comment (Dumplingz) [inline tests/action_management/test_get_actions.c:68]: Should this test be deleted?

Comment (isaacfink21) [reply tests/action_management/test_get_actions.c:68]: Which version are you looking at? The new version should be changed to `use_on->kind`

Comment (Dumplingz) [reply tests/action_management/test_get_actions.c:68]: Ah, I missed that you’ve actually replaced use with use_on. Then this part looks good

File: src/action_management/src/get_actions.c (hunk 1 above)
Comment (isaacfink21) [reply src/action_management/src/get_actions.c:15]: Which files are you referring to? I couldn't find any labeled as "BIN"

Comment (Dumplingz) [reply src/action_management/src/get_actions.c:15]: They’re hidden files; I found them in the “files changed” tab on this issue

Comment (isaacfink21) [reply src/action_management/src/get_actions.c:15]: Oh I see them now - thanks for pointing that out. They should be removed now :)

File: src/action_management/src/get_actions.c (hunk 2 above)
Comment (isaacfink21) [reply src/action_management/src/get_actions.c:18]: The goal of this PR is just to make these new actions legal—their actual implementations will be handled by [cli](https://github.com/uchicago-cs/chiventure/blob/3a2fd911dac8aede02818f2a522b7d23dd97694a/src/cli/src/operations.c) (see line 162) and [game-state](https://github.com/uchicago-cs/chiventure/blob/87ecb1d8dd7fe7f8183fa55cfe7472f740d714cc/src/game-state/src/game_action.c). Some actions, like `WALK`, already had implementations but were made illegal, and will now be fully functional. The other ones will need to be implemented separately - I can add an issue to the backlog once this PR goes through :)

Comment (Dumplingz) [reply src/action_management/src/get_actions.c:18]: Sounds good. Don't see anything else that needs changes then!

File: include/action_management/action_structs.h [hunk 4]
```diff
@@ -18,15 +18,20 @@ enum actions {
...
     // KIND 2 ACTIONS - ACTION <path>
-    GO, // synonyms: "walk"
+    GO,
+    WALK,
 
     /* KIND 3 ACTIONS - ACTION <item_item> */
-    USE,
+    USE_ON,
     PUT
 };
```
Comment (MaxineK36) [inline include/action_management/action_structs.h:36]: Could you add some comments here indicating which legal actions currently have implementations? 
Also, as of right now, what happens if you try to write one of these actions in a game file? Relatedly, what happens if you try to use one of these actions?

Comment (isaacfink21) [reply include/action_management/action_structs.h:38]: Currently, legal actions without implementation will display a message that the action cannot be used with a given object.

E.g. upon trying to pickup a table in examples/test.wdl:
```
> pickup chair
   Action PICKUP can't be requested with item CHAIR
```

**Comment by isaacfink21** [comment]
PR has been approved! Action synonyms have been re-introduced as legal for the following actions:
```
/* KIND 2: <PATH> */
GO, // synonym: WALK
/* KIND 1: <ITEM> */
CONSUME // synonyms: USE, EAT, DRINK
```
The `WALK` synonym is now fully functional, and the other three synonyms are legal but require implementation from CLI. As there are now two versions of the 'use' action, the original version (kind `ITEM_ITEM`) has also been refactored to `USE_ON`.

**Comment by MaxineK36** [comment]
Issue Score: ✔️+

Comments:
This is a decently solid first PR! A few things to think about for next time:
1. When you create a PR, you are requesting a review of the code in the branch you want to merge. As such, you should not push any commits to the branch right after creating the PR; you should only do so in response to a code review. Make sure that, when you create a PR, the code is ready for review and passes all the tests.
2. The summary should have been updated when the goal of the PR changed.

**Pull request merged**